### PR TITLE
Coats

### PR DESCRIPTION
**Pull request by kol1th**
Here they are do what you will, pm me if anything is missing

### CODE REVIEW

File: code/modules/clothing/suits/toggles.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/obj/item/clothing/suit/wintercoat/verb/togglehood()
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: We could use this in a different way, and across a larger area.

Action buttons like with the jetpack

File: code/modules/clothing/suits/toggles.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	set name = "Toggle Hood"
+	set category = "Object"
+	set src in usr
+	if(usr.incapacitated())
+		return
+	else
+		var/mob/living/carbon/human/user = usr
+		if(!usr || !istype(usr) || !user.head) // the head slot is empty
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: !user.head would return true if the user is not wearing anything, try user.head instead

File: code/modules/clothing/suits/toggles.dm [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			to_chat(usr, "You try to put your hood up but something is in the way.")
+			return
+		if(src.is_hooded == 2)
+			equip(/obj/item/clothing/head/winterhood, SLOT_HEAD)
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: Change this for equip_to_slot(/obj/item/clothing/head/winterhood, slot_head)

File: code/modules/clothing/suits/toggles.dm [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+			icon_state = "[initial(icon_state)]_t"
+			user.head = /obj/item/clothing/head/winterhood
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: This would be handled by equip_to_slot

File: code/modules/clothing/suits/miscellaneous.dm [hunk 5]
```diff
@@ -490,3 +490,66 @@
...
+	desc = "A heavy jacket made from 'synthetic' animal furs."
+	icon_state = "coatwinter"
+	item_state = "labcoat"
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
+	heat_conductivity = SPACESUIT_HEAT_CONDUCTIVITY
+	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 10, rad = 0)
+	action_button_name = "Toggle Hood"
+	var/is_hooded = 2
+
+/obj/item/clothing/head/winterhood
```
Comment (Skullyton) [inline code/modules/clothing/suits/miscellaneous.dm]: If you want this to be just something that's equipped to the person, you may want to add canremove = 0 so they can't just keep taking it off and dropping it on the floor

File: code/modules/clothing/suits/miscellaneous.dm [hunk 6]
```diff
@@ -490,3 +490,66 @@
...
+/obj/item/clothing/suit/wintercoat
+	name = "winter coat"
+	desc = "A heavy jacket made from 'synthetic' animal furs."
+	icon_state = "coatwinter"
+	item_state = "labcoat"
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
+	heat_conductivity = SPACESUIT_HEAT_CONDUCTIVITY
+	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 10, rad = 0)
+	action_button_name = "Toggle Hood"
+	var/is_hooded = 2
```
Comment (Skullyton) [inline code/modules/clothing/suits/miscellaneous.dm]: Why start at 2 and then go to 1? Why not keep it boolean and 1 and 0?

File: code/modules/clothing/suits/toggles.dm [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+			flags = initial(flags)
+			body_parts_covered &= ~(HEAD)
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: Why the flag changes? Wouldn't the hood cover it?

File: code/modules/clothing/suits/toggles.dm [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+			to_chat(usr, "You put \the hood up.")
+			src.is_hooded = 1
+		else
+			icon_state = "[initial(icon_state)]"
+			to_chat(usr, "You put \the hood down.")
+			flags = 0
```
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: Nowhere in this else statement does it say that you're unequipping the hood.

u_equip(usr.head,0) would drop it and presumably nullspace it

File: code/modules/clothing/suits/toggles.dm (hunk 1 above)
Comment (Skullyton) [inline code/modules/clothing/suits/toggles.dm]: To do an action button, you just add

/obj/item/clothing/suit/wintercoat/ui_action_click()
	togglehood()

**Comment by kol1th** [comment]
All I can currently find still wrong is: the hood doesn't leave when you unequip the coat.

**Pull request merged**